contracts/auctions/src/storage.rs

```diff
@@ -0,0 +1,290 @@
+use soroban_sdk::{contracttype, log, panic_with_error, vec, Address, Env, Vec};
+
+use crate::error::ContractError;
+
+// Values used to extend the TTL of storage
+pub const DAY_IN_LEDGERS: u32 = 17280;
+pub const BUMP_AMOUNT: u32 = 7 * DAY_IN_LEDGERS;
+pub const LIFETIME_THRESHOLD: u32 = BUMP_AMOUNT - DAY_IN_LEDGERS;
+
+// consts for Pagination
+// since we start counting from 1, default would be 1 as well
+pub const DEFAULT_INDEX: u32 = 1;
+pub const DEFAULT_LIMIT: u32 = 10;
+
+#[contracttype]
+#[derive(Clone)]
+pub enum DataKey {
+    Admin,
+    IsInitialized,
+    AuctionId,
+    AllAuctions,
+    HighestBid(u64),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[contracttype]
+pub struct ItemInfo {
+    pub collection_addr: Address,
+    pub item_id: u64,
+    pub minimum_price: Option<u64>,
+    pub buy_now_price: Option<u64>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[contracttype]
+pub struct Auction {
+    pub id: u64,
+    pub item_info: ItemInfo,
+    pub seller: Address,
+    pub highest_bid: Option<u64>,
+    pub highest_bidder: Address,
+    pub end_time: u64,
+    pub status: AuctionStatus,
+    pub currency: Address,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+#[contracttype]
+pub struct HighestBid {
+    pub bid: u64,
+    pub bidder: Address,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+#[contracttype]
+pub enum AuctionStatus {
+    Active,
+    Ended,
+    Cancelled,
+    Paused,
+}
+
+pub fn generate_auction_id(env: &Env) -> Result<u64, ContractError> {
+    let id = env
+        .storage()
+        .instance()
+        .get::<_, u64>(&DataKey::AuctionId)
+        .unwrap_or_default()
+        + 1u64;
+    env.storage().instance().set(&DataKey::AuctionId, &id);
+    env.storage()
+        .instance()
+        .extend_ttl(LIFETIME_THRESHOLD, BUMP_AMOUNT);
+
+    Ok(id)
+}
+
+pub fn get_auctions(
+    env: &Env,
+    start_index: Option<u32>,
+    limit: Option<u32>,
+) -> Result<Vec<Auction>, ContractError> {
+    let start_index = start_index.unwrap_or(DEFAULT_INDEX);
+    let limit = limit.unwrap_or(DEFAULT_LIMIT);
+
+    let mut auctions = vec![&env];
+
+    for id in start_index..=limit {
+        match get_auction_by_id(env, id as u64) {
+            Ok(auction) => auctions.push_back(auction),
+            Err(ContractError::AuctionNotFound) => break,
```

Comment:
```suggestion
            Err(ContractError::AuctionNotFound) => continue,
```
Why stopping the whole loop? You are not returning the error anyway.